jobserver/models/core.py

```diff
@@ -1033,6 +1043,37 @@ def rotate_token(self):
         # return the unhashed token so it can be passed to a consuming service
         return token
 
+    class BadLoginToken(Exception):
```

Comment:
This section reminds me of a conversation @ghickman and I were having yesterday where we were talking about how and when to refactor parts of job-server. I wonder if the "business logic" of generating and validating a login token (and arguably a PAT token) would be better off in a separate module? The User class has seen a few changes recently and is just starting to look a bit too big.

Reply:
If we have a convention/location of where that should go, I can move it. 

But possilby as a follow up?

Reply:
`commands` for me clashes with django management commands, conceptually.

I've previously used a `logic` package name for this?